packages/library-legacy-sham/src/transaction.ts

```diff
@@ -63,6 +71,72 @@ export class Transaction {
         const addressString = publicKey.toBase58();
         this.#tx = setTransactionFeePayer(addressString as Address, this.#tx);
     }
+    get lastValidBlockHeight(): number | undefined {
+        if (
+            'lifetimeConstraint' in this.#tx &&
+            this.#tx.lifetimeConstraint != null &&
+            typeof this.#tx.lifetimeConstraint === 'object' &&
+            'blockhash' in this.#tx.lifetimeConstraint
+        ) {
+            return Number((this.#tx as ITransactionWithBlockhashLifetime).lifetimeConstraint.lastValidBlockHeight);
+        }
+    }
+    set lastValidBlockHeight(slot: number | null | undefined) {
+        this.#tx = setTransactionLifetimeUsingBlockhash(
+            {
+                // Use the existing value or set it to `undefined` otherwise.
+                blockhash:
+                    'lifetimeConstraint' in this.#tx &&
+                    this.#tx.lifetimeConstraint != null &&
+                    typeof this.#tx.lifetimeConstraint === 'object' &&
+                    'blockhash' in this.#tx.lifetimeConstraint
+                        ? (this.#tx as ITransactionWithBlockhashLifetime).lifetimeConstraint.blockhash
+                        : (undefined as unknown as Blockhash),
+                lastValidBlockHeight: BigInt(slot ?? Number.MAX_SAFE_INTEGER),
+            },
+            this.#tx
+        );
+    }
+    get nonceInfo() {
+        throw getUnimplementedError('Transaction#nonceInfo (getter)');
```

Comment:
Didn't feel like implementing this, because we'd have to implement `TransactionInstruction`. I'll wait until this is necessary.